### ISSUE TITLE
Worker crashed/SIGSEGV while closing Sushi the cat

### ISSUE THREAD

**Issue by trichimtrich** [giongto35/cloud-game#180]
### Env
Dev
MacOS

### Step to produce
1. Run coordinator + connect worker
2. Run `Sushi the cat` game
3. Wait for 5 secs
4. Close the browser tab

### Expected
No crash

### Output
<details>

```
I0503 08:05:32.739968   19154 main.go:29] Initializing worker server
I0503 08:05:32.740110   19154 worker.go:47] Starting monitoring server for overwork
I0503 08:05:32.740162   19154 monitoring.go:53] Starting monitoring server at :6601
I0503 08:05:32.740169   19154 monitoring.go:67] Prometheus metric is enabled at :6601/worker/metrics
I0503 08:05:32.742265   19154 storage.go:25] Warn: Failed to create client: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
I0503 08:05:32.742323   19154 worker.go:154] Listening at port: localhost: 9000
I0503 08:05:32.742361   19154 handlers.go:95] Worker connecting to coordinator: ws://localhost:8000/wso?zone=
I0503 08:05:32.744018   19154 handlers.go:72] Connected to coordinator successfully. &{0xc00007aa40} <nil>
I0503 08:05:32.744422   19154 coordinator.go:45] Received serverID  f1b830c3-b63b-4cc5-bf87-ced43eb4d5ee
I0503 08:05:34.028183   19154 worker.go:75] &{0xc00011d900 0xc000152300 {} 0x4093440 false false false false 0xc00038c1c0 {0xc0003ba0e0 map[] false false} map[Access-Control-Allow-Origin:[*]] true 0 -1 0 false false [] 0 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [0 0 0 0 0 0 0 0 0 0] [0 0 0] 0xc000394070 0} echo
I0503 08:05:34.066044   19154 coordinator.go:57] Received a request to createOffer from browser via coordinator
I0503 08:05:34.066145   19154 webrtc.go:111] === StartClient ===
I0503 08:05:34.091253   19154 webrtc.go:131] Add video track
I0503 08:05:34.091313   19154 webrtc.go:142] Add audio track
I0503 08:05:34.091607   19154 webrtc.go:200] Created Offer
I0503 08:05:34.091814   19154 webrtc.go:181] OnIceCandidate: candidate:foundation 1 udp 2130706431 10.254.254.4 58142 typ host
I0503 08:05:34.091884   19154 coordinator.go:89] Start peerconnection c8560466-ca6c-4cd4-8ed9-0c06f57367a2
I0503 08:05:34.092021   19154 webrtc.go:181] OnIceCandidate: candidate:foundation 1 udp 2130706431 192.168.1.203 49402 typ host
I0503 08:05:34.092089   19154 webrtc.go:181] OnIceCandidate: candidate:foundation 1 udp 1694498815 3.0.84.67 63572 typ srflx raddr 0.0.0.0 rport 63572
I0503 08:05:34.112148   19154 coordinator.go:106] Received answer SDP from browser
I0503 08:05:34.112337   19154 webrtc.go:233] Set Remote Description
I0503 08:05:34.114649   19154 webrtc.go:165] ICE Connection State has changed: checking
I0503 08:05:34.120404   19154 coordinator.go:127] Received remote Ice Candidate from browser
I0503 08:05:34.120443   19154 webrtc.go:244] Decoded Ice: candidate:5 1 TCP 2105524479 cb291d6e-ca6a-49f1-b738-6b004923aa8a.local 9 typ host tcptype active
I0503 08:05:34.120510   19154 webrtc.go:252] Add Ice Candidate: candidate:5 1 TCP 2105524479 cb291d6e-ca6a-49f1-b738-6b004923aa8a.local 9 typ host tcptype active
I0503 08:05:34.120669   19154 coordinator.go:127] Received remote Ice Candidate from browser
I0503 08:05:34.120700   19154 webrtc.go:244] Decoded Ice: candidate:0 1 UDP 2122252543 cb291d6e-ca6a-49f1-b738-6b004923aa8a.local 65246 typ host
I0503 08:05:34.120742   19154 webrtc.go:252] Add Ice Candidate: candidate:0 1 UDP 2122252543 cb291d6e-ca6a-49f1-b738-6b004923aa8a.local 65246 typ host
I0503 08:05:34.129238   19154 coordinator.go:127] Received remote Ice Candidate from browser
I0503 08:05:34.129319   19154 webrtc.go:244] Decoded Ice: candidate:1 1 UDP 1686052863 3.0.84.67 65246 typ srflx raddr 0.0.0.0 rport 0
I0503 08:05:34.129565   19154 webrtc.go:252] Add Ice Candidate: candidate:1 1 UDP 1686052863 3.0.84.67 65246 typ srflx raddr 0.0.0.0 rport 0
I0503 08:05:38.133692   19154 webrtc.go:165] ICE Connection State has changed: connected
I0503 08:05:38.133739   19154 webrtc.go:169] ConnectionStateConnected
I0503 08:05:38.133754   19154 webrtc.go:282] Start streaming
I0503 08:05:38.206274   19154 webrtc.go:149] Data channel 'game-input'-'824633971448' open.
I0503 08:05:39.728844   19154 coordinator.go:150] Received a start request from coordinator
I0503 08:05:39.730081   19154 coordinator.go:301] Starting game Sushi The Cat
I0503 08:05:39.730126   19154 coordinator.go:307] Got Room from local  <nil>  ID:
I0503 08:05:39.730181   19154 room.go:73] Init new room:  b33469b2a237ab2___Sushi The Cat Sushi The Cat {Sushi The Cat assets/games/Sushi The Cat.gba gba}
I0503 08:05:39.730218   19154 room.go:197] Updated player Index to:  0
I0503 08:05:39.730231   19154 coordinator.go:325] Is PC in room false
I0503 08:05:39.730237   19154 handlers.go:126] Detach peerconnection
I0503 08:05:39.730747   19154 room.go:92] Check  /Users/something/.cr/save/b33469b2a237ab2___Sushi The Cat.dat  on online storage :  false
I0503 08:05:39.730823   19154 room.go:325] Check if game is on cloud storage
I0503 08:05:39.731141   19154 room.go:98] Room b33469b2a237ab2___Sushi The Cat started. GamePath: assets/games/Sushi The Cat.gba, GameName: Sushi The Cat
Video pixel: {0 0 0 2 0 2} 2 0 1 2Libretro API version: 1
ROM size: 540296
  library_name: mGBA
  library_version: 0.8 429d2f2
  valid_extensions: gba|gb|gbc
  need_fullpath: false
  block_extract: false
[Env]: get variable: mgba_gb_model
[Env]: get variable: mgba_use_bios
[Env]: get variable: mgba_skip_bios
[Env]: get variable: mgba_sgb_borders
[Env]: get variable: mgba_frameskip
[Env]: get variable: mgba_idle_optimization
[Env]: get variable: mgba_color_correction
-----------------------------------
--- System audio and video info ---
-----------------------------------
  Aspect ratio:  1.5
  Base width:  240
  Base height:  160
  Max width:  240
  Max height:  160
  Sample rate:  32768
  FPS:  59.727500915527344
-----------------------------------
I0503 08:05:39.767574   19154 room.go:120] Viewport custom size is disabled, base size will be used instead 240x160
I0503 08:05:39.767684   19154 room.go:129] meta:  {assets/emulator/libretro/cores/mgba_libretro 240 160 32768 59 240 160 1.5}
I0503 08:05:39.767760   19154 media.go:114] Video Encoder:  VP8
I0503 08:05:39.767846   19154 media.go:49] Enter fan audio
listening audio channel true
I0503 08:05:51.008673   19154 coordinator.go:271] Received a terminate session  c8560466-ca6c-4cd4-8ed9-0c06f57367a2
I0503 08:05:51.008883   19154 webrtc.go:263] ===StopClient===
I0503 08:05:51.010060   19154 webrtc.go:159] Data channel closed
I0503 08:05:51.010090   19154 webrtc.go:160] Closed webrtc
I0503 08:05:51.010166   19154 handlers.go:126] Detach peerconnection
I0503 08:05:51.010247   19154 room.go:228] Cleaning session:  6e797475-74bc-4310-804c-101c482b2d47
I0503 08:05:51.010857   19154 room.go:231] found session:  6e797475-74bc-4310-804c-101c482b2d47
I0503 08:05:51.010964   19154 room.go:234] Removed session  6e797475-74bc-4310-804c-101c482b2d47  from room:  b33469b2a237ab2___Sushi The Cat
I0503 08:05:51.010984   19154 handlers.go:136] No session in room
I0503 08:05:51.010991   19154 room.go:264] Closing room and director of room  b33469b2a237ab2___Sushi The Cat
I0503 08:05:51.010996   19154 room.go:268] Saved Game before closing room
I0503 08:05:51.011025   19154 room.go:279] Closing input of room  b33469b2a237ab2___Sushi The Cat
I0503 08:05:51.011416   19154 handlers.go:139] Signal input chan
I0503 08:05:51.011500   19154 room.go:223] Peerconn done
I0503 08:05:51.011534   19154 webrtc.go:165] ICE Connection State has changed: closed
I0503 08:05:51.033474   19154 media.go:70] Room  b33469b2a237ab2___Sushi The Cat  audio channel closed
I0503 08:05:51.033688   19154 media.go:156] Room  b33469b2a237ab2___Sushi The Cat  video channel closed
I0503 08:05:51.033703   19154 encoder.go:160] Releasing encoder
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1450 pc=0x47e37b8]

runtime stack:
runtime.throw(0x4a0c9fb, 0x2a)
	/usr/local/go/src/runtime/panic.go:1116 +0x72
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:679 +0x46a

goroutine 48 [syscall]:
runtime.cgocall(0x4712590, 0xc0001c56b0, 0xc000152168)
	/usr/local/go/src/runtime/cgocall.go:133 +0x5b fp=0xc0001c5680 sp=0xc0001c5648 pc=0x4005afb
github.com/giongto35/cloud-game/pkg/encoder/vpx-encoder._Cfunc_vpx_codec_encode(0xc000152130, 0xc000152168, 0x29b, 0x1, 0x0, 0x1, 0x0)
	_cgo_gotypes.go:314 +0x4d fp=0xc0001c56b0 sp=0xc0001c5680 pc=0x459695d
github.com/giongto35/cloud-game/pkg/encoder/vpx-encoder.(*VpxEncoder).startLooping.func3(0xc000152100, 0xc000000000, 0x1)
	/Users/something/Desktop/share/cloud-game/pkg/encoder/vpx-encoder/encoder.go:139 +0xf4 fp=0xc0001c5710 sp=0xc0001c56b0 pc=0x4597ae4
github.com/giongto35/cloud-game/pkg/encoder/vpx-encoder.(*VpxEncoder).startLooping(0xc000152100)
	/Users/something/Desktop/share/cloud-game/pkg/encoder/vpx-encoder/encoder.go:139 +0x1ca fp=0xc0001c57d8 sp=0xc0001c5710 pc=0x459729a
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc0001c57e0 sp=0xc0001c57d8 pc=0x4069491
created by github.com/giongto35/cloud-game/pkg/encoder/vpx-encoder.(*VpxEncoder).init
	/Users/something/Desktop/share/cloud-game/pkg/encoder/vpx-encoder/encoder.go:114 +0x187

goroutine 1 [chan receive]:
main.main()
	/Users/something/Desktop/share/cloud-game/cmd/worker/main.go:40 +0x557

goroutine 7 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x50a8dc0)
	/Users/something/Desktop/share/gopath/pkg/mod/github.com/golang/glog@v0.0.0-20160126235308-23def4e6c14b/glog.go:882 +0x8b
created by github.com/golang/glog.init.0
	/Users/something/Desktop/share/gopath/pkg/mod/github.com/golang/glog@v0.0.0-20160126235308-23def4e6c14b/glog.go:410 +0x26f

goroutine 18 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc0000a8730)
	/Users/something/Desktop/share/gopath/pkg/mod/go.opencensus.io@v0.22.0/stats/view/worker.go:154 +0x100
created by go.opencensus.io/stats/view.init.0
	/Users/something/Desktop/share/gopath/pkg/mod/go.opencensus.io@v0.22.0/stats/view/worker.go:32 +0x57

goroutine 19 [IO wait]:
internal/poll.runtime_pollWait(0x572ce58, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:203 +0x55
internal/poll.(*pollDesc).wait(0xc0003b4098, 0x72, 0x0, 0x0, 0x49ec711)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Accept(0xc0003b4080, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:384 +0x1d4
net.(*netFD).accept(0xc0003b4080, 0xb21605df5b6a34a2, 0x21fd5b6a34a2, 0x100000001)
	/usr/local/go/src/net/fd_unix.go:238 +0x42
net.(*TCPListener).accept(0xc000388040, 0x5eae0ace, 0xc00005ec88, 0x4089e16)
	/usr/local/go/src/net/tcpsock_posix.go:139 +0x32
net.(*TCPListener).Accept(0xc000388040, 0xc00005ecd8, 0x18, 0xc00008a600, 0x42e5bcc)
	/usr/local/go/src/net/tcpsock.go:261 +0x64
net/http.(*Server).Serve(0xc0003ba000, 0x4b10640, 0xc000388040, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2901 +0x25d
net/http.(*Server).ListenAndServe(0xc0003ba000, 0x49ea76b, 0x1)
	/usr/local/go/src/net/http/server.go:2830 +0xb7
github.com/giongto35/cloud-game/pkg/worker.(*Worker).spawnServer(0xc0000f6300, 0x2328)
	/Users/something/Desktop/share/cloud-game/pkg/worker/worker.go:135 +0x276
github.com/giongto35/cloud-game/pkg/worker.(*Worker).initializeWorker(0xc0000f6300)
	/Users/something/Desktop/share/cloud-game/pkg/worker/worker.go:168 +0x235
created by github.com/giongto35/cloud-game/pkg/worker.(*Worker).Run
	/Users/something/Desktop/share/cloud-game/pkg/worker/worker.go:41 +0x3f

goroutine 20 [IO wait]:
internal/poll.runtime_pollWait(0x572cf38, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:203 +0x55
internal/poll.(*pollDesc).wait(0xc000156298, 0x72, 0x0, 0x0, 0x49ec711)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Accept(0xc000156280, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:384 +0x1d4
net.(*netFD).accept(0xc000156280, 0xc000064c88, 0xc000088000, 0x5396108)
	/usr/local/go/src/net/fd_unix.go:238 +0x42
net.(*TCPListener).accept(0xc00000f540, 0xc000064cc8, 0x400f8c8, 0x30)
	/usr/local/go/src/net/tcpsock_posix.go:139 +0x32
net.(*TCPListener).Accept(0xc00000f540, 0x4973960, 0xc00016a120, 0x48cfaa0, 0x5092e70)
	/usr/local/go/src/net/tcpsock.go:261 +0x64
net/http.(*Server).Serve(0xc000168000, 0x4b10640, 0xc00000f540, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2901 +0x25d
net/http.(*Server).ListenAndServe(0xc000168000, 0xc00003c590, 0xf)
	/usr/local/go/src/net/http/server.go:2830 +0xb7
github.com/giongto35/cloud-game/pkg/monitoring.(*ServerMonitoring).Run(0xc0000c4500, 0x0, 0xc00004cfb8)
	/Users/something/Desktop/share/cloud-game/pkg/monitoring/monitoring.go:71 +0x239
github.com/giongto35/cloud-game/pkg/worker.(*Worker).RunMonitoringServer(0xc0000f6300)
	/Users/something/Desktop/share/cloud-game/pkg/worker/worker.go:48 +0x87
created by github.com/giongto35/cloud-game/pkg/worker.(*Worker).Run
	/Users/something/Desktop/share/cloud-game/pkg/worker/worker.go:42 +0x61

goroutine 21 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:144 +0x96
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.Notify.func1
	/usr/local/go/src/os/signal/signal.go:127 +0x44

goroutine 14 [chan receive]:
github.com/giongto35/cloud-game/pkg/cws.(*Client).Heartbeat(0xc00007aa40)
	/Users/something/Desktop/share/cloud-game/pkg/cws/cws.go:139 +0xc2
created by github.com/giongto35/cloud-game/pkg/worker.(*Handler).Run
	/Users/something/Desktop/share/cloud-game/pkg/worker/handlers.go:73 +0x175

goroutine 36 [IO wait]:
internal/poll.runtime_pollWait(0x572cd78, 0x72, 0xffffffffffffffff)
	/usr/local/go/src/runtime/netpoll.go:203 +0x55
internal/poll.(*pollDesc).wait(0xc0000f6598, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc0000f6580, 0xc000170000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:169 +0x201
net.(*netFD).Read(0xc0000f6580, 0xc000170000, 0x1000, 0x1000, 0x5, 0x6, 0x198)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc0000100f8, 0xc000170000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:184 +0x8e
bufio.(*Reader).fill(0xc00007ccc0)
	/usr/local/go/src/bufio/bufio.go:100 +0x103
bufio.(*Reader).Peek(0xc00007ccc0, 0x2, 0xc0003d186d, 0xc00016a480, 0xc00012b050, 0x4890340, 0xc0000966c0)
	/usr/local/go/src/bufio/bufio.go:138 +0x4f
github.com/gorilla/websocket.(*Conn).read(0xc00013ab00, 0x2, 0x40563dc, 0x406afd0, 0xc000185c78, 0x4ffc5066, 0xc000055400)
	/Users/something/Desktop/share/gopath/pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:357 +0x40
github.com/gorilla/websocket.(*Conn).advanceFrame(0xc00013ab00, 0x0, 0x0, 0x4a817c401)
	/Users/something/Desktop/share/gopath/pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:787 +0x5c
github.com/gorilla/websocket.(*Conn).NextReader(0xc00013ab00, 0xbfa3a09cc0ed9838, 0x8e9be48d3, 0x50a8b00, 0x72, 0x0)
	/Users/something/Desktop/share/gopath/pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:947 +0x8f
github.com/gorilla/websocket.(*Conn).ReadMessage(0xc00013ab00, 0xbfa3a09cc0ed9838, 0x8e9be48d3, 0x50a8b00, 0x0, 0x0, 0x50a8b00)
	/Users/something/Desktop/share/gopath/pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:1028 +0x2f
github.com/giongto35/cloud-game/pkg/cws.(*Client).Listen(0xc00007aa40)
	/Users/something/Desktop/share/cloud-game/pkg/cws/cws.go:153 +0xa8
github.com/giongto35/cloud-game/pkg/worker.(*Handler).Run(0xc0003b2000)
	/Users/something/Desktop/share/cloud-game/pkg/worker/handlers.go:75 +0x19d
created by github.com/giongto35/cloud-game/pkg/worker.(*Worker).initializeWorker
	/Users/something/Desktop/share/cloud-game/pkg/worker/worker.go:150 +0xe6

goroutine 109 [runnable]:
sync.runtime_SemacquireMutex(0xc00003d3cc, 0x1, 0x1)
	/usr/local/go/src/runtime/sema.go:71 +0x47
sync.(*Mutex).lockSlow(0xc00003d3c8)
	/usr/local/go/src/sync/mutex.go:138 +0xfc
sync.(*Mutex).Lock(...)
	/usr/local/go/src/sync/mutex.go:81
github.com/giongto35/cloud-game/pkg/emulator/libretro/nanoarch.(*naEmulator).GetLock(...)
	/Users/something/Desktop/share/cloud-game/pkg/emulator/libretro/nanoarch/savestates.go:25
github.com/giongto35/cloud-game/pkg/emulator/libretro/nanoarch.(*naEmulator).Save(0xc00012e790, 0x0, 0x0)
	/Users/something/Desktop/share/cloud-game/pkg/emulator/libretro/nanoarch/savestates.go:38 +0x1d2
github.com/giongto35/cloud-game/pkg/emulator/libretro/nanoarch.(*naEmulator).SaveGame(0xc00012e790, 0xc00009df80, 0x4aff1a0, 0x403fd16)
	/Users/something/Desktop/share/cloud-game/pkg/emulator/libretro/nanoarch/naemulator.go:172 +0x48
github.com/giongto35/cloud-game/pkg/worker/room.(*Room).SaveGame(0xc0000f7480, 0xc0000a9130, 0x4a26010)
	/Users/something/Desktop/share/cloud-game/pkg/worker/room/room.go:316 +0x69
github.com/giongto35/cloud-game/pkg/worker/room.(*Room).Close.func1(0xc0000f7480)
	/Users/something/Desktop/share/cloud-game/pkg/worker/room/room.go:273 +0x2b
created by github.com/giongto35/cloud-game/pkg/worker/room.(*Room).Close
	/Users/something/Desktop/share/cloud-game/pkg/worker/room/room.go:271 +0x118

goroutine 16 [IO wait]:
internal/poll.runtime_pollWait(0x572cc98, 0x72, 0xffffffffffffffff)
	/usr/local/go/src/runtime/netpoll.go:203 +0x55
internal/poll.(*pollDesc).wait(0xc000156518, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc000156500, 0xc00017e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:169 +0x201
net.(*netFD).Read(0xc000156500, 0xc00017e000, 0x1000, 0x1000, 0x408995a, 0x21fe857cb4ae, 0x0)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc000010158, 0xc00017e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:184 +0x8e
net/http.(*connReader).Read(0xc00016a660, 0xc00017e000, 0x1000, 0x1000, 0xbfa3a0b181b04428, 0x1c3db4a3a7, 0x50a8b00)
	/usr/local/go/src/net/http/server.go:786 +0xf4
bufio.(*Reader).fill(0xc00007cea0)
	/usr/local/go/src/bufio/bufio.go:100 +0x103
bufio.(*Reader).Peek(0xc00007cea0, 0x4, 0x1c3db4a3a7, 0x50a8b00, 0x0, 0x0, 0x50a8b00)
	/usr/local/go/src/bufio/bufio.go:138 +0x4f
net/http.(*conn).serve(0xc00011d900, 0x4b12a40, 0xc00007aa80)
	/usr/local/go/src/net/http/server.go:1920 +0x9d7
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2933 +0x35c

goroutine 100 [runnable]:
sync.runtime_Semrelease(0xc00003d3cc, 0xc0001cec01, 0x1)
	/usr/local/go/src/runtime/sema.go:66 +0x3e
sync.(*Mutex).unlockSlow(0xc00003d3c8, 0xc00000000c)
	/usr/local/go/src/sync/mutex.go:224 +0xb3
sync.(*Mutex).Unlock(...)
	/usr/local/go/src/sync/mutex.go:190
github.com/giongto35/cloud-game/pkg/emulator/libretro/nanoarch.(*naEmulator).ReleaseLock(...)
	/Users/something/Desktop/share/cloud-game/pkg/emulator/libretro/nanoarch/savestates.go:30
github.com/giongto35/cloud-game/pkg/emulator/libretro/nanoarch.(*naEmulator).Start(0xc00012e790)
	/Users/something/Desktop/share/cloud-game/pkg/emulator/libretro/nanoarch/naemulator.go:161 +0x10a
github.com/giongto35/cloud-game/pkg/worker/room.NewRoom.func1(0xc0000f7480, 0xc000521680, 0x2260, 0x7ffeefbffa53, 0xe, 0x1, 0x0, 0x140, 0xf0, 0x49ea758, ...)
	/Users/something/Desktop/share/cloud-game/pkg/worker/room/room.go:134 +0x7cf
created by github.com/giongto35/cloud-game/pkg/worker/room.NewRoom
	/Users/something/Desktop/share/cloud-game/pkg/worker/room/room.go:89 +0x3d1
exit status 2
```
</details>

**Comment by giongto35** [comment]
saw it few times before. I will check it. Seems like it happened at VPX flow.

@trichimtrich can we swap gstreamer here?

**Comment by trichimtrich** [comment]
Hmm, I'm not sure about `gstreamer`

Didn't find any good plugin that fit for images streaming from gstreamer
A close one is in this post https://stackoverflow.com/questions/59452619/create-movie-from-images-using-gstreamer
But only send 1 freezed frame

Another approach is `ffmpeg` like this https://stackoverflow.com/questions/3063695/streaming-video-using-stream-of-images
But it needs tweak a lot to fit in our use-case.

Both work in higher level (wrapper) via pipeline (with raw image) and generate a stream like what we did here in low level (write directly to VP8/H264 stream).

So I dont think we should replace current workflow

**Comment by 88hcsif** [comment]
I think this is a race condition while trying to free the VP8 encoder resources while a frame is being encoded.
I think a similar race condition can also affect the H264 encoder.
I think we just need to coordinate the <code>release</code> and the <code>startLooping</code> functions.
I gave it a go but I think I ended up not always freeing up the resources which may cause a memory leak so I have not nailed a fix yet...
